### PR TITLE
Update modals mnrva 567

### PR DESCRIPTION
**Pull request by lovedeep92**
**JIRA:**
See ticket here - https://jira.rax.io/browse/MNRVA-567

**Description:**

1) Replace <hx-modal> with new common component <app-modal> and added every where.

2) create functionality to pass header, message according to your choice in <app-modal> tag.

3) Complete delete event functionality with common modal.

**Testing:**

`npm run test:unit`

 ## Screenshots:
(if applicable)

### CODE REVIEW

**Review by pras7430 — COMMENTED** [review]
it is open with Master and 82 files
Screenshot 2020-09-08 at 7 04 23 PM
 as changes

File: src/app/_shared/components/modal/modal.component.ts [hunk 1]
```diff
@@ -21,8 +23,12 @@ export class ModalComponent implements OnInit {
   ngOnInit(): void {
   }
 
-  close() {
+  close(message) {
```
Comment (pras7430) [inline src/app/_shared/components/modal/modal.component.ts:26]: if we are using only two value then boolean may fill our purpose, the comparison will not be required  everywhere

**Review by pras7430 — APPROVED** [review]

File: projects/admin/src/app/tab/_features/resources/pages/details/admin-resource-details.page.ts [hunk 2]
```diff
@@ -43,6 +46,7 @@ export class AdminResourceDetailsPage implements OnInit {
   // TODO(optional): attempt to move this logic to a route resolve as opposed
   // to connecting the subscription to the request within the component
   ngOnInit() {
+    this.message = "Are you sure you'd like to delete this Resource?";
```
[the remark this answers is not in the document]
Comment (lovedeep92) [reply projects/admin/src/app/tab/_features/resources/pages/details/admin-resource-details.page.ts:49]: done .. have added during declaration!

File: projects/admin/src/app/tab/_features/monitors/pages/details/details.component.ts [hunk 3]
```diff
@@ -87,6 +91,7 @@ export class DetailsComponent implements OnInit {
   }
 
   ngOnInit() {
+    this.message = "Are you sure you'd like to delete this Monitor?";
```
[the remark this answers is not in the document]
Comment (lovedeep92) [reply projects/admin/src/app/tab/_features/monitors/pages/details/details.component.ts:94]: done .. have added during declaration!

File: src/app/_features/monitors/pages/details/monitor-details.page.ts [hunk 4]
```diff
@@ -93,7 +95,6 @@ export class MonitorDetailsPage implements OnInit {
 
   ngOnInit() {
     this.message = "Are you sure you'd like to delete this Monitor?";
```
[the remark this answers is not in the document]
Comment (lovedeep92) [reply src/app/_features/monitors/pages/details/monitor-details.page.ts:97]: done .. have added during declaration!

File: src/app/_services/events/events.service.ts [hunk 5]
```diff
@@ -78,4 +78,23 @@ export class EventsService {
         })
       )
   }
+
+    /**
+   * @description
+   * @param id string
+   */
+  deleteEvent(id:string) {
+      console.log("id ", id);
```
[the remark this answers is not in the document]
Comment (lovedeep92) [reply src/app/_services/events/events.service.ts:87]: done!!

File: src/app/_features/events/components/list/eventslist.component.ts [hunk 6]
```diff
@@ -11,14 +11,24 @@ import { Router } from '@angular/router';
   styleUrls: ['./eventslist.component.scss']
 })
 export class EventslistComponent implements OnInit {
-  
+
+  @ViewChild('delEventLink') delEvent:ElementRef;
   events:Event[];
   size:number = 1;
+  selectedEvents: any = [];
+  isDeleted : boolean = true;
```
[the remark this answers is not in the document]
Comment (lovedeep92) [reply src/app/_features/events/components/list/eventslist.component.ts:19]: I am using the same standard we were using in other files!

File: src/app/_shared/components/modal/modal.component.spec.ts [hunk 7]
```diff
@@ -42,15 +42,15 @@ describe('ModalComponent', () => {
   });
 
   it('should show id', () => {
-    component.id = "devUbuntu";
+    component.header = "delete devUbuntu";
```
[the remark this answers is not in the document]
Comment (lovedeep92) [reply src/app/_shared/components/modal/modal.component.spec.ts:45]: I have modified test case and fetching value using query and check if the value is same with what I initialise above!

File: src/app/_shared/components/modal/modal.component.ts (hunk 1 above)
Comment (lovedeep92) [reply src/app/_shared/components/modal/modal.component.ts:26]: changed it to flag with true false value!

**Pull request merged**